synapse/storage/schema/main/delta/72/04event_backfill_access_time.sql

```diff
@@ -0,0 +1,27 @@
+/* Copyright 2022 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+-- Add a table that keeps track of when we last tried to backfill an event. This
+-- allows us to be more intelligent when we decide to retry (we don't need to
+-- fail over and over) and we can process that event in the background so we
+-- don't block on it each time.
+CREATE TABLE IF NOT EXISTS event_backfill_attempts(
+    event_id TEXT NOT NULL,
```

Comment:
Do we need the `room_id` here as well? I think `event_id` is unique enough (there is some caveat here that I can't recall)?

Reply:
Event IDs are not necessarily unique to a room. Only in recent room versions, where they are the hash of the redacted form of the event, can they be considered practically unique enough, but since we still have to worry about old room versions, then they're not ...

⇒ `room_id` required

Reply:
Thanks for the info!

Also looked harder at the `events` table and found that its primary key is `stream_ordering` 😮

Reply:
More info and discussion at https://github.com/matrix-org/synapse/pull/13589#discussion_r961116999

Reply:
From the linked discussion, summarized the outcome in https://github.com/matrix-org/synapse/pull/13701